node/impl/full/eth.go

```diff
@@ -234,7 +234,37 @@ func (a *EthModule) EthGetBlockByHash(ctx context.Context, blkHash ethtypes.EthH
 	return newEthBlockFromFilecoinTipSet(ctx, ts, fullTxInfo, a.Chain, a.StateAPI)
 }
 
-func (a *EthModule) parseBlkParam(ctx context.Context, blkParam string, strict bool) (tipset *types.TipSet, err error) {
+// blkParamToTipsetAfter returns the next non-nil tipset after the given block param.
+func (a *EthModule) blkParamToTipsetAfter(ctx context.Context, blkParam string) (*types.TipSet, error) {
+	switch blkParam {
+	case "earliest", "pending":
```

Comment:
Alternatively, we can reject "pending" in some specific cases, rather than allow it in some specific cases.